### PR TITLE
tests using junit4

### PR DESCRIPTION
**Pull request by nhachicha**
- [x] Switching to the Junit 4 runner
- [x] Template of running tests on the UI thread
- [x] Introducing the usage of custom Rules with @RunTestInLooperThread
- [x] Removed non running tests (Mixtypes etc)

### CODE REVIEW

**Comment by nhachicha** [comment]
retest this please

**Comment by cmelchior** [comment]
retest this please

**Comment by nhachicha** [comment]
@realm/java

File: realm/realm-library/src/androidTest/java/io/realm/RealmTest.java [hunk 1]
```diff
@@ -1815,27 +1913,103 @@ public void testMutableMethodsOutsideTransactions() throws JSONException, IOExce
...
-        try { testRealm.createOrUpdateObjectFromJson(AllTypesPrimaryKey.class, jsonObjStr);     fail(); } catch (IllegalStateException expected) {}
-        try { testRealm.createOrUpdateObjectFromJson(AllTypesPrimaryKey.class, jsonObjStream2); fail(); } catch (IllegalStateException expected) {}
-
-        try { testRealm.createAllFromJson(AllTypesPrimaryKey.class, jsonArr);                   fail(); } catch (RealmException expected) {}
-        try { testRealm.createAllFromJson(AllTypesPrimaryKey.class, jsonArrStr);                fail(); } catch (RealmException expected) {}
-        try { testRealm.createAllFromJson(NoPrimaryKeyNullTypes.class, jsonArrStream);             fail(); } catch (IllegalStateException expected) {}
-        try { testRealm.createOrUpdateAllFromJson(AllTypesPrimaryKey.class, jsonArr);           fail(); } catch (RealmException expected) {}
-        try { testRealm.createOrUpdateAllFromJson(AllTypesPrimaryKey.class, jsonArrStr);        fail(); } catch (RealmException expected) {}
-        try { testRealm.createOrUpdateAllFromJson(AllTypesPrimaryKey.class, jsonArrStream2);    fail(); } catch (IllegalStateException expected) {}
+        try {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/RealmTest.java]: Can we get the old layout back. This looks horrible.

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunTestInLooperThread.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+package io.realm.rule;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+@Target(METHOD) @Retention(RUNTIME)
+public @interface RunTestInLooperThread {
```
Comment (cmelchior) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunTestInLooperThread.java]: Add a comment about where/when this can be used

**Comment by cmelchior** [comment]
Nice. Besides a few comments about style/comments missing. Big :+1:

**Comment by cmelchior** [comment]
:+1:

**Comment by beeender** [comment]
:+1:

**Comment by zaki50** [comment]
:+1:

**Comment by nhachicha** [comment]
squashed

**Pull request merged**